envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterFactory.kt

```diff
@@ -114,25 +114,36 @@ class RBACFilterFactory(
             loggedEndpointsPolicies.values
         )
 
-        val shadowPolicies = commonPolicies + loggedEndpointsPolicies
-        val actualPolicies = commonPolicies + allowUnlistedPolicies
+        val shadowPolicies = (commonPolicies + loggedEndpointsPolicies)
```

Comment:
This method looks clearer to me like this:
```kotlin
    private fun getRules(
        incomingPermissions: Incoming,
        snapshot: GlobalSnapshot,
        roles: List<Role>
    ): Rules {

        val incomingEndpointsPolicies = getIncomingEndpointPolicies(
                incomingPermissions, snapshot, roles
        )

        val restrictedEndpointsPolicies = incomingEndpointsPolicies.asSequence()
            .filter { it.endpoint.unlistedClientsPolicy == Incoming.UnlistedPolicy.BLOCKANDLOG }
            .map { (endpoint, policy) -> "$endpoint" to policy }.toMap()

        val loggedEndpointsPolicies = incomingEndpointsPolicies.asSequence()
            .filter { it.endpoint.unlistedClientsPolicy == Incoming.UnlistedPolicy.LOG }
            .map { (endpoint, policy) -> "$endpoint" to policy }.toMap()

        val allowUnlistedPolicies = unlistedAndLoggedEndpointsPolicies(
            incomingPermissions,
            (statusRoutePolicy + restrictedEndpointsPolicies).values,
            loggedEndpointsPolicies.values
        )

        val restrictedEndpointsPoliciesWithFullAccessClient = restrictedEndpointsPolicies.mapValues {
            addFullAccessClients(it.value)
        }

        val shadowPolicies = (statusRoutePolicy + restrictedEndpointsPolicies + loggedEndpointsPolicies)
            .map { (endpoint, policy) -> endpoint to policy.build() }.toMap()
        val actualPolicies = (statusRoutePolicy + restrictedEndpointsPoliciesWithFullAccessClient + allowUnlistedPolicies)
            .map { (endpoint, policy) -> endpoint to policy.build() }.toMap()

        val actualRules = RBAC.newBuilder()
            .setAction(RBAC.Action.ALLOW)
            .putAllPolicies(actualPolicies)
            .build()

        val shadowRules = RBAC.newBuilder()
            .setAction(RBAC.Action.ALLOW)
            .putAllPolicies(shadowPolicies)
            .build()
        // build needs to be called before any modifications happen so we do not have to clone it

        return Rules(shadowRules = shadowRules, actualRules = actualRules)
    }
```
A common policy is no longer common and IMO explicitly stating what policies are going where is more readable. Also, I moved `builds` around so they diff better from a previous version.